### PR TITLE
Update some of the Slice compiler keyword lists.

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
This PR updates the keywords that the Slice compiler will escape when generating code for C++, C#, Java, PHP, Python, and Swift.

MATLAB, Ruby, and Javascript still need checking.

A question for Javascript though, should we be escaping typescript keywords/types as well? Right now it seems like we only escape pure Javascript keywords, and it's not clear that that's correct.

----

Sources:
https://en.cppreference.com/w/cpp/keyword
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/
https://en.wikipedia.org/wiki/List_of_Java_keywords
https://www.php.net/manual/en/reserved.keywords.php
https://docs.python.org/3/reference/lexical_analysis.html#keywords
https://docs.swift.org/swift-book/documentation/the-swift-programming-language/lexicalstructure/

Also, found some doc-bugs in the official Swift book. Turns out keeping documentation up to date is hard.
https://github.com/swiftlang/swift-book/pull/331
https://github.com/swiftlang/swift-book/pull/330

### CODE REVIEW

File: cpp/src/slice2cs/CsUtil.cpp [hunk 1]
```diff
@@ -31,21 +31,21 @@ namespace
...
+            "params",   "private",   "protected", "public",  "readonly",  "record",   "ref",        "required",
+            "return",   "sbyte",     "scoped",    "sealed",  "short",     "sizeof",   "stackalloc", "static",
+            "string",   "struct",    "switch",    "this",    "throw",     "true",     "try",        "typeof",
+            "uint",     "ulong",     "unchecked", "unsafe",  "ushort",    "using",    "var",        "virtual",
+            "void",     "volatile",  "while"};
         bool found = binary_search(
             &keywordList[0],
             &keywordList[sizeof(keywordList) / sizeof(*keywordList)],
-            name,
-            Slice::CICompare());
```
Comment (InsertCreativityHere) [inline cpp/src/slice2cs/CsUtil.cpp:48]: There are no uppercase keywords in C#, so there's no need to pass our custom comparator.
In the other languages, we only pass it when it's necessary.

**Comment by InsertCreativityHere** [comment]
`clang-format` is very picky whether these lists are organized in a grid or one big list.
It's not clear what criterion is uses for switching between these two,
but it doesn't seem to be purely based on the number of elements.

I think we should consider disabling clang-format for this function,
and keeping these as one big list, instead of in a grid. Sure, the grid looks nice
but it's impossible to review, since git just sees everything changing.

**Review by bernardnormier — APPROVED** [review]
We should not case-insentive comparison for case-sensitive languages.

**Comment by pepone** [comment]
> A question for Javascript though, should we be escaping typescript keywords/types as well? Right now it seems like we only escape pure Javascript keywords, and it's not clear that that's correct.

I think we should stick to JavaScript only. TypeScript extends JavaScript and by definition all JavaScript code is valid in TypeScript.

**Review by externl — APPROVED** [review]

**Comment by externl** [comment]
The build is currently failing!

**Comment by InsertCreativityHere** [comment]
That's what I get for editing the file directly in the browser : vP

**Pull request merged**